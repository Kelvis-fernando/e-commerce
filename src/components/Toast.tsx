import React, { useEffect, useState } from "react";
import { ToastContainer, ToastCloseButton, Progress } from "../styles/toast";
import { ToastProps } from "../types/toast";
import { CheckCircle, WarningCircle, X } from "phosphor-react";

const Toast: React.FC<ToastProps> = ({
  message,
  duration = 3000,
  onClose,
  typeOfToast = "success",
}) => {
  const [progress, setProgress] = useState(0);

  useEffect(() => {
    let timer: NodeJS.Timeout;
    const step = 10;
    const increment = (step / duration) * 100;
    timer = setInterval(() => {
      setProgress((prevProgress) => {
        const newProgress = prevProgress + increment;
        if (newProgress >= 100) {
          clearInterval(timer);
          onClose();
          window.location.reload();
        }
        return newProgress;
      });
    }, step);
    return () => clearInterval(timer);
  }, [duration, onClose]);

  return (
    <ToastContainer>
      {typeOfToast === "success" ? (
        <CheckCircle size={32} style={{ color: "#2D5731" }} />
      ) : (
        <WarningCircle size={32} style={{ color: "#BD2A2E" }} />
      )}
      <span>{message}</span>
      <Progress style={{ width: `${progress}%` }} />
      <ToastCloseButton onClick={onClose}>
        <X size={20} />
      </ToastCloseButton>
    </ToastContainer>
  );
};

export default Toast;
