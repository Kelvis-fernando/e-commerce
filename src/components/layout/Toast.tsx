import { useEffect, useState } from "react";
import { ToastContainer } from "../../styles/layout/toast";
import { ToastProps } from "../../types/toast";

const Toast = ({ title, message, duration = 1 }: ToastProps) => {
  const [progress, setProgress] = useState(0);

  useEffect(() => {
    const interval = setInterval(() => {
      setProgress((prevProgress) => prevProgress + 1);
    }, duration / 100);

    return () => {
      clearInterval(interval);
    };
  }, [duration]);

  return (
    <ToastContainer>
      <h1>{title}</h1>
      <p>{message}</p>
      {progress}
    </ToastContainer>
  );
};

export default Toast;
