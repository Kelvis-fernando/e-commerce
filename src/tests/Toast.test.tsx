import React from "react";
import { render } from "@testing-library/react";
import Toast from "../components/Toast";

describe("Toast", () => {
  test("should render and match snapshot", () => {
    const { container } = render(
      <Toast
        message={""}
        onClose={function (): void {
          throw new Error("Function not implemented.");
        }}
      />
    );

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
