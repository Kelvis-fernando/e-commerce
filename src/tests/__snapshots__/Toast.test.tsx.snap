// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe earMgI"
  >
    <svg
      fill="currentColor"
      height="32"
      style="color: rgb(45, 87, 49);"
      viewBox="0 0 256 256"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <polyline
        fill="none"
        points="172 104 113.3 160 84 132"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
      <circle
        cx="128"
        cy="128"
        fill="none"
        r="96"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
    </svg>
    <span />
    <div
      class="sc-dkrFOg eJClsy"
      style="width: 0%;"
    />
    <button
      class="sc-gswNZR hQuefH"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="56"
          y1="56"
          y2="200"
        />
        <line
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="56"
          y1="200"
          y2="56"
        />
      </svg>
    </button>
  </div>
</div>
`;
